test/common/testlib.py

```diff
@@ -743,27 +761,32 @@ def setUp(self):
         if opts.address and self.provision is not None:
             raise unittest.SkipTest("Cannot provision multiple machines if a specific machine address is specified")
 
-        self.machine = None
         self.browser = None
         self.machines = {}
         provision = self.provision or {'machine1': {}}
 
-        # First create all machines, wait for them later
-        for key in sorted(provision.keys()):
-            options = provision[key].copy()
-            if 'address' in options:
-                del options['address']
-            if 'dns' in options:
-                del options['dns']
-            if 'dhcp' in options:
-                del options['dhcp']
-            machine = self.new_machine(**options)
-            self.machines[key] = machine
-            if not self.machine:
-                self.machine = machine
-            if opts.trace:
-                print("Starting {0} {1}".format(key, machine.label))
-            machine.start()
+        test_method = getattr(self.__class__, self._testMethodName)
+        if getattr(test_method, "_testlib__non_destructive", False) and not opts.address:
+            self.machines['machine1'] = MachineCase.get_global_machine()
```

Comment:
hm I'd say more just error out in that case. Now it'll have opts.address if it's being called from run-tests; but if the file itself is called the provision check won't raise, so we could add the extra check to raise in case we're calling the `check-*` file directly and someone added a provision on a @nondestructive.